Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.kt

```diff
@@ -0,0 +1,151 @@
+        viewModel.close()
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        _binding = EditTagBinding.inflate(LayoutInflater.from(context))
+        return buildDialog()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+        _dialogEditTag = null
+    }
+
+    private fun buildDialog(): Dialog {
+        val context: Context = ContextThemeWrapper(requireContext(), R.style.Dialog)
+        _dialogEditTag = AlertDialog.Builder(context)
+            .setView(binding.root)
+            .setTitle(R.string.rename_tag)
+            .setNegativeButton(R.string.cancel, null)
+            .setPositiveButton(R.string.save, null)
+            .create()
+
+        dialogEditTag.setOnShowListener(this)
+        binding.inputTagName.editText?.doAfterTextChanged { s -> viewModel.updateUiState(s.toString()) }
+
+        return dialogEditTag
+    }
+
+    private fun setObservers() {
+        viewModel.uiState.observe(this, { (_, _, errorMsg) ->
+            if (errorMsg != null) {
+                binding.inputTagName.error = getString(errorMsg)
+                dialogEditTag.getButton(DialogInterface.BUTTON_POSITIVE).isEnabled = false
+            } else {
+                binding.inputTagName.error = null
+                dialogEditTag.getButton(DialogInterface.BUTTON_POSITIVE).isEnabled = true
+            }
+        })
+
+        viewModel.event.observe(this, { event: TagDialogEvent ->
+            when (event) {
+                CloseEvent, FinishEvent -> dismiss()
+                ShowErrorEvent -> {
+                    val context = requireContext()
+                    DialogUtils.showDialogWithEmail(context, context.getString(R.string.rename_tag_message))
+                }
+                is ConflictEvent -> {
+                    val (canonicalTagName, oldTagName) = event
+                    showDialogErrorConflict(canonicalTagName, oldTagName)
+                }
+            }
+        })
+    }
+
+    override fun onShow(dialog: DialogInterface) {
+        setObservers()
+        startUiState()
+        showDialogRenameTag()
+    }
+
+    private fun startUiState() {
+        viewModel.start(tag)
+        binding.inputTagName.editText?.setText(tag.name)
+    }
+
+    /**
+     * Change the dialog layout to show a conflict message in which the user can merge two tags or cancel the merge
+     */
+    private fun showDialogErrorConflict(canonical: String, tagOld: String) {
+        binding.message.text = getString(R.string.dialog_tag_conflict_message, canonical, tagOld, canonical)
+        binding.message.visibility = View.VISIBLE
+        binding.inputTagName.visibility = View.GONE
+
+        dialogEditTag.setTitle(R.string.dialog_tag_conflict_title)
+
+        val positiveButton = dialogEditTag.getButton(DialogInterface.BUTTON_POSITIVE)
+        positiveButton.text = getString(R.string.dialog_tag_conflict_button_positive)
```

Comment:
Question (❓): What if we were explicit with the `positiveButton` visibility as well, just like we do for the `negativeButton` and `neutralButton`?

PS: Same apply for the `showDialogRenameTag()` function below.

Reply:
For the moment is not needed since the positive button is always visible, but to avoid any bugs in future changes, I agree that making it explicit would make this code less error-prone. Fixed in 045bddff1a9612962d432649dced6d1fddc31c4e.

Reply:
You are a rockstar, thank you! 🙏